src/core_atmosphere/dynamics/mpas_atm_iau.F

```diff
@@ -69,6 +69,7 @@ real (kind=RKIND) function atm_iau_coef(configs, itimestep, dt) result(wgt_iau)
       
          if(itimestep <= nsteps_iau) then
             !wgt_iau = 1./nsteps_iau     
+            !wgt_iau = nint(dt /time_window_sec), which will be divided by dt when applied later.
```

Comment:
At least to my untrained eyes, adding the two commented lines L71-72 cause confusion.  It would be good to have one comment with the analytical equation, and another comment explaining how that is executed in the code if it is not clear from the analytical equation itself.  It should be clear enough that somebody coming back to refactor the code later (e.g., for time-varying weights) understands what the original intention is.

Reply:
Thanks for your suggestion, JJ. I had left those comments for a possible extension for 4D_IAU later, but just removed them all now for clarity. Please take a look to see if it makes sense to you.

Reply:
It's OK to keep commented lines for 4D IAU if there is also a comment about why the extra lines are there, such as `TODO(SH): use this line in 4D IAU`.

Reply:
I also added new comments to explain how the weight function was derived and applied, trying to provide the necessary info for possible updates for 4D IAU later. If you need more clarifications, please let me know. Thanks.